src/backend/PortabilityService.AnalysisService/IServiceSettings.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+//TODO (yumeng): uncommend next line
+//using Microsoft.Fx.Portability.Github;
+using Microsoft.WindowsAzure.Storage;
+using System;
+
+namespace PortabilityService.AnalysisService
+{
+    /// <summary>
+    /// Portability service settings
+    /// </summary>
+    public interface IServiceSettings
+    {
+        /// <summary>
+        /// How often to check for updates for data stored in Azure Storage.
+        /// Used to check for updates for catalog.bin and fxmember.json.gz
+        /// </summary>
+        TimeSpan UpdateFrequency { get; }
```

Comment:
This could be simpler--instead of updating the data while the service is live, we can redeploy the service. The orchestrator should make this cheap and orderly.

Reply:
Agree. Though we probably should do similar Retry-After in the DataCatalog service client instead as this will replaced soon.